src/client/interpreter/locators/index.ts

```diff
@@ -0,0 +1,61 @@
+"use strict";
+import * as _ from 'lodash';
+import { fixInterpreterPath, fixInterpreterDisplayName } from './helpers';
+import { IInterpreterLocatorService, PythonInterpreter } from '../contracts';
+import { InterpreterVersionService } from '../interpreterVersion';
+import { IS_WINDOWS, Is_64Bit, arePathsSame, areBasePathsSame } from '../../common/utils';
+import { RegistryImplementation } from '../../common/registry';
+import { CondaEnvService } from './services/condaEnvService';
+import { VirtualEnvService, getKnownSearchPathsForVirtualEnvs } from './services/virtualEnvService';
+import { KnownPathsService, getKnownSearchPathsForInterpreters } from './services/KnownPathsService';
+import { CurrentPathService } from './services/currentPathService';
+import { WindowsRegistryService } from './services/windowsRegistryService';
+import { VirtualEnvironmentManager } from '../virtualEnvs';
+import { CondaEnvFileService, getEnvironmentsFile as getCondaEnvFile } from './services/condaEnvFileService';
+
+export class PythonInterpreterLocatorService implements IInterpreterLocatorService {
+    private interpreters: PythonInterpreter[] = [];
+    private locators: IInterpreterLocatorService[] = [];
+    constructor(private virtualEnvMgr: VirtualEnvironmentManager) {
+        const versionService = new InterpreterVersionService();
+        // The order of the services is important
```

Comment:
Is not ending comments in a period a thing in the TypeScript world? (Serious question because it's definitely a thing to always use a period in Python.) Otherwise I would definitely use an exclamation point here. 😄 (And if you don't have a preference then I vote for putting in periods to know that a comment definitely is finished and not accidentally incomplete.)

Reply:
>Is not ending comments in a period a thing in the TypeScript world

Never heard of this.

Reply:
Now i have